sys/net/gnrc/link_layer/lorawan/gnrc_lorawan.c

```diff
@@ -71,6 +71,9 @@ void gnrc_lorawan_set_rx2_dr(gnrc_lorawan_t *mac, uint8_t rx2_dr)
 
 void gnrc_lorawan_set_adr(gnrc_lorawan_t *mac, bool adr)
 {
+    if ((adr == true) && (mac->mlme.adr == false)) {
+        mac->mlme.adr_ack_cnt = 0;
+    }
     mac->mlme.adr = adr;
```

Comment:
```suggestion
    if (mac->mlme.adr == adr) {
        return -EALREADY;
    }
    mac->mlme.adr = adr;
    mac ->mlme.adr_ack_cnt = 0;
```

It doesn't hurt to set it to 0 again and probably saves some ROM.
This would mean the function should be `int` and the upper layer should take care of the error code (as we usually do with NETOPTS)